lib/DxcSupport/FileIOHelper.cpp

```diff
@@ -136,8 +136,7 @@ void ReadBinaryFile(IMalloc *pMalloc, LPCWSTR pFileName, void **ppData,
   // Open file
   std::ifstream ifs(CW2A(pFileName).m_psz, std::ios::binary|std::ios::ate);
   if(ifs.fail()) {
-    fprintf(stderr, "The system cannot file the file specified:");
-    throw ::hlsl::Exception(GetLastError());
+    IFT(HRESULT_FROM_WIN32(GetLastError()));
```

Comment:
This macro confuses me. If I understand it correctly, if it's already an error, it leaves it be. if it isn't, it will assign a facility of 7, and negate it. Either way it fails so I guess you get an exception, which seems right. I guess I just don't understand why not just throw the errno. Because it's not negative so might not be seen as a failure?

Reply:
yeah, if the errno was a negative number, I'd just throw it. This way it looks just like the win-version of the code see code (e.g. line 106). The facility being added and then stripped away is unnecessary as you said, but just trying to use existing code.

Reply:
Yeah. I understand that the windows code was this way. I was just concerned that since we were throwing unix codes instead of windows it might bite us later on, but it seems to work okay where it converts this to a string in dxcapi.use.cpp. So I suppose we're good. At any rate. we're no worse off than before.